src/silo/config/append_config.cpp

```diff
@@ -0,0 +1,113 @@
+#include "silo/config/append_config.h"
+
+#include "config/config_interface.h"
+#include "config/source/yaml_file.h"
+#include "silo/common/json_type_definitions.h"
+#include "silo/config/config_defaults.h"
+
+using silo::config::ConfigKeyPath;
+using silo::config::YamlFile;
+
+namespace {
+ConfigKeyPath appendConfigOptionKey() {
+   return YamlFile::stringToConfigKeyPath("appendConfig");
+}
+ConfigKeyPath siloDirectoryOptionKey() {
+   return YamlFile::stringToConfigKeyPath("siloDirectory");
+}
+ConfigKeyPath appendFileOptionKey() {
+   return YamlFile::stringToConfigKeyPath("appendFile");
+}
+ConfigKeyPath partitionIdOptionKey() {
+   return YamlFile::stringToConfigKeyPath("partitionId");
+}
+ConfigKeyPath siloDataSourceOptionKey() {
+   return YamlFile::stringToConfigKeyPath("siloDataSource");
+}
+ConfigKeyPath dataVersionOptionKey() {
+   return YamlFile::stringToConfigKeyPath("dataVersion");
+}
+}  // namespace
+
+namespace silo::config {
+
+void AppendConfig::validate() const {}
+
+AppendConfig AppendConfig::withDefaults() {
+   AppendConfig result;
+   result.overwriteFrom(getConfigSpecification().getConfigSourceFromDefaults());
+   return result;
+}
+
+ConfigSpecification AppendConfig::getConfigSpecification() {
+   return ConfigSpecification{
+      .program_name = "silo append",
+      .attribute_specifications{
+         ConfigAttributeSpecification::createWithDefault(
+            siloDirectoryOptionKey(),
+            ConfigValue::fromPath("."),
+            "The path to a silo-directory, a directory that contains silo outputs. This may be "
+            "used for input (see ..) and will be used for the output of the new silo state"
+         ),
+         ConfigAttributeSpecification::createWithoutDefault(
+            dataVersionOptionKey(),
+            ConfigValueType::STRING,
+            "The data version of the new database. If not given, a new data version will be minted "
+            "instead."
```

Comment:
I think it would be good to give some advice on when to use which. What's the consequence of specifying / not specifying it?